benches/advanced_ot_based_ecdsa.rs

```diff
@@ -0,0 +1,231 @@
+use criterion::{criterion_group, Criterion};
+use frost_secp256k1::VerifyingKey;
+use rand::Rng;
+use rand_core::SeedableRng;
+
+mod bench_utils;
+use crate::bench_utils::{
+    ot_ecdsa_prepare_presign, ot_ecdsa_prepare_sign, ot_ecdsa_prepare_triples,
+    PreparedSimulatedSig, MAX_MALICIOUS,
+};
+
+use threshold_signatures::{
+    ecdsa::{
+        ot_based_ecdsa::{
+            presign::presign,
+            sign::sign,
+            triples::{generate_triple_many, TriplePub, TripleShare},
+            PresignArguments, PresignOutput,
+        },
+        SignatureOption,
+    },
+    participants::Participant,
+    protocol::Protocol,
+    test_utils::{
+        create_rngs, run_protocol, run_protocol_and_take_snapshots, run_simulated_protocol,
+        MockCryptoRng, Simulator,
+    },
+};
+
+type PreparedSimulatedTriples = (
+    Participant,
+    Box<dyn Protocol<Output = Vec<(TripleShare, TriplePub)>>>,
+    Simulator,
+);
+
+type PreparedSimulatedPresig = (
+    Participant,
+    Box<dyn Protocol<Output = PresignOutput>>,
+    Simulator,
+);
+
+fn threshold() -> usize {
+    *MAX_MALICIOUS + 1
+}
+
+fn participants_num() -> usize {
+    *MAX_MALICIOUS + 1
+}
+
+/// Benches the triples protocol
+fn bench_triples(c: &mut Criterion) {
+    let num = participants_num();
+    let max_malicious = *MAX_MALICIOUS;
+    let mut group = c.benchmark_group("triples");
+    group.measurement_time(std::time::Duration::from_secs(200));
+
+    group.bench_function(
+        format!("ot_ecdsa_triples_advanced_MAX_MALICIOUS_{max_malicious}_PARTICIPANTS_{num}"),
+        |b| {
+            b.iter_batched(
+                || prepare_simulated_triples(num),
+                |(rparticipant, rprot, sprot)| run_simulated_protocol(rparticipant, rprot, sprot),
+                criterion::BatchSize::SmallInput,
+            );
+        },
+    );
+}
+
+/// Benches the presigning protocol
+fn bench_presign(c: &mut Criterion) {
+    let num = participants_num();
+    let max_malicious = *MAX_MALICIOUS;
+    let mut group = c.benchmark_group("presign");
+    group.measurement_time(std::time::Duration::from_secs(300));
+
+    let mut rng = MockCryptoRng::seed_from_u64(42);
+    let rngs = create_rngs(num, &mut rng);
+    let (protocols, _) = ot_ecdsa_prepare_triples(num, threshold(), &rngs, &mut rng);
+    let two_triples = run_protocol(protocols).expect("Running triple preparations should succeed");
+
+    group.bench_function(
+        format!("ot_ecdsa_presign_advanced_MAX_MALICIOUS_{max_malicious}_PARTICIPANTS_{num}"),
+        |b| {
+            b.iter_batched(
+                || prepare_simulated_presign(&two_triples),
+                |(rparticipant, rprot, sprot)| run_simulated_protocol(rparticipant, rprot, sprot),
+                criterion::BatchSize::SmallInput,
+            );
+        },
+    );
+}
+
+/// Benches the signing protocol
+fn bench_sign(c: &mut Criterion) {
```

Comment:
Nit:
With more descriptive names it's clear without the doc what the function does.
```suggestion
fn bench_signing_protocol(c: &mut Criterion) {
```

Reply:
I have only added the doc to prevent clippy from yelling. I do not think what you proposed gives additional information.